src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -0,0 +1,599 @@
+import io.confluent.kafkarest.entities.ConsumerAssignmentResponse;
+import io.confluent.kafkarest.entities.ConsumerCommittedRequest;
+import io.confluent.kafkarest.entities.ConsumerCommittedResponse;
+import io.confluent.kafkarest.entities.TopicPartitionOffsetMetadata;
+
+import io.confluent.rest.exceptions.RestException;
+import io.confluent.rest.exceptions.RestNotFoundException;
+import io.confluent.rest.exceptions.RestServerErrorException;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+
+import io.confluent.kafkarest.*; //compiling v1 versions
+
+/**
+ * Manages consumer instances by mapping instance IDs to consumer objects, processing read requests,
+ * and cleaning up when consumers disappear.
+ */
+public class KafkaConsumerManager {
+
+  private static final Logger log = LoggerFactory.getLogger(KafkaConsumerManager.class);
+
+  private final KafkaRestConfig config;
+  private final Time time;
+  //private final String zookeeperConnect;
+  private final String bootstrapServers;    
+  private final MetadataObserver mdObserver;
+  private final int iteratorTimeoutMs;
+
+  // KafkaConsumerState is generic, but we store them untyped here. This allows many operations to
+  // work without having to know the types for the consumer, only requiring type information
+  // during read operations.
+  private final Map<ConsumerInstanceId, KafkaConsumerState> consumers =
+      new HashMap<ConsumerInstanceId, KafkaConsumerState>();
+  // Read operations are common and there may be many concurrently, so they are farmed out to
+  // worker threads that can efficiently interleave the operations. Currently we're just using a
+  // simple round-robin scheduler.
+  private final List<KafkaConsumerWorker> workers;
+  private final AtomicInteger nextWorker;
+  // A few other operations, like commit offsets and closing a consumer can't be interleaved, but
+  // they're also comparatively rare. These are executed serially in a dedicated thread.
+  private final ExecutorService executor;
+  private KafkaConsumerFactory consumerFactory;
+  private final PriorityQueue<KafkaConsumerState> consumersByExpiration =
+      new PriorityQueue<KafkaConsumerState>();
+  private final ExpirationThread expirationThread;
+
+  public KafkaConsumerManager(KafkaRestConfig config, MetadataObserver mdObserver) {
+    this.config = config;
+    this.time = config.getTime();
+    //this.zookeeperConnect = config.getString(KafkaRestConfig.ZOOKEEPER_CONNECT_CONFIG);
+    this.bootstrapServers = config.getString(KafkaRestConfig.BOOTSTRAP_SERVERS_CONFIG);
+    this.mdObserver = mdObserver;
+    this.iteratorTimeoutMs = config.getInt(KafkaRestConfig.CONSUMER_ITERATOR_TIMEOUT_MS_CONFIG);
```

Comment:
Presumably we can get rid of this field since we're not relying on the old consumer iterator semantics anymore?